### PR TITLE
Fixes a manifest validation and lets add PWAs without icon

### PR DESCRIPTION
**Pull request by juliantoledo**

### CODE REVIEW

**Comment by juliantoledo** [comment]
Fixes #304

**Comment by juliantoledo** [comment]
@ithinkihaveacat @sebastianbenz PTAL

**Comment by sebastianbenz** [comment]
Can you update the tests as well?

**Comment by juliantoledo** [comment]
Of course, I just added one manifest without icon, I will also add another test with a invalid manifest.

**Comment by juliantoledo** [comment]
done

File: test/manifests/no-icon.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "Twitter",
+  "short_name": "Twitter",
+  "start_url": "/?utm_source=web_app_manifest",
+  "display": "standalone",
+  "gcm_sender_id": "49625052041",
```
Comment (sebastianbenz) [inline test/manifests/no-icon.json]: nit: to harden test cases against future changes, I'd recommend limiting test data to the minimum needed to expose a specific behavior. In this case this'd mean removing: gcm_sender_id, gcm_user_visible_only, related_applications,...

File: test/manifests/invalid.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (sebastianbenz) [inline test/manifests/invalid.json]: nit: you could explain why this manifest is invalid via the filename (I actually don't know why this is invalid)

File: test/lib/pwa.js [hunk 3]
```diff
@@ -39,23 +39,39 @@ const MANIFEST_URL = 'https://www.terra.com.br/manifest-br.json';
...
         pwa = new Pwa(MANIFEST_URL, manifest);
         pwa.id = '123456789';
         dataFetcher.readFile(LIGHTHOUSE_JSON_EXAMPLE)
           .then(data => {
             lighthouse = new Lighthouse('123456789', 'www.domain.com', JSON.parse(data));
-            done();
           });
-      });
+      })],
+      [dataFetcher.readFile(MANIFEST_DATA_NO_ICON)
```
Comment (sebastianbenz) [inline test/lib/pwa.js]: Is it possible to directly create the PWA obj? The advantages would be:

* faster tests
* more maintainable tests as they have less dependencies to other APIs (e.g. dataFetcher)

File: test/manifests/no-icon.json (hunk 1 above)
Comment (juliantoledo) [inline test/manifests/no-icon.json]: sure, sounds good.

File: test/lib/pwa.js (hunk 3 above)
Comment (juliantoledo) [inline test/lib/pwa.js]: I think so, I had a hard time escaping manifest properly to embed them in the test. I will try again.

Comment (juliantoledo) [inline test/lib/pwa.js]: @sebastianbenz added the manifest as inline text and replaced the datafetcher for a regular fs call to read the lighthouse json.

**Comment by juliantoledo** [comment]
PTAL

File: test/lib/pwa.js [hunk 4]
```diff
@@ -13,50 +13,61 @@
...
-        pwa = new Pwa(MANIFEST_URL, manifest);
-        pwa.id = '123456789';
-        dataFetcher.readFile(LIGHTHOUSE_JSON_EXAMPLE)
-          .then(data => {
-            lighthouse = new Lighthouse('123456789', 'www.domain.com', JSON.parse(data));
-            done();
-          });
-      });
-  });
+  const manifest = new Manifest(MANIFEST_URL, JSON.parse(MANIFEST_DATA));
```
Comment (sebastianbenz) [inline test/lib/pwa.js]: You can avoid the parse step if you don't declare the manifest as a string:

```
const MANIFEST_DATA = {
  name: "Test",
  ...
}

```

File: test/lib/pwa.js [hunk 5]
```diff
@@ -13,50 +13,61 @@
...
+  const pwa = new Pwa(MANIFEST_URL, manifest);
+  pwa.id = '123456789';
+  const manifestNoIcon = new Manifest(MANIFEST_URL, JSON.parse(MANIFEST_NO_ICON));
+  const pwaNoIcon = new Pwa(MANIFEST_URL, manifestNoIcon);
```
Comment (sebastianbenz) [inline test/lib/pwa.js]: I've created a helper method for instantiating the PWAs [here](https://github.com/GoogleChrome/gulliver/blob/master/test/models/pwa.js#L129). We could use this everytime we create a new pwa in a test. The benefit is that we only need to touch the helper method if for example the PWA constructor API changes.

File: test/lib/pwa.js (hunk 4 above)
Comment (juliantoledo) [inline test/lib/pwa.js]: I wanted to have the text as similar as possible to the real manifest, that is why I was reading them from files.

**Comment by juliantoledo** [comment]
Fixed the latest comments.
@sebastianbenz PTAL

**Pull request merged**